types/itemsjs/index.d.ts

```diff
@@ -125,26 +138,40 @@ declare namespace itemsjs {
     }
 
     /** Configuration for itemsjs */
-    interface Configuration<I extends {}, S extends string, A extends keyof I & string> {
+    interface Configuration<
+        I extends {},
+        S extends string,
+        A extends keyof I & string,
+        IdField extends string = "id",
```

Comment:
Would it cause any errors to also enforce `keyof I` here?

```suggestion
        IdField extends keyof I & string = "id",
```

Reply:
Yes, indeed, it won't work because of the default value.

I initially tried `IdField extends keyof I & string = "id"` but it causes a TypeScript error because at that point, `keyof I` can be evaluated as `never`. So `"id"` isn't assignable to `never`.

To avoid that, I used `IdField extends string = "id"` and rely on the usage of `I[IdField]` elsewhere to enforce correctness. Let me know if you'd prefer an alternative approach!

But to be honest, I'm not sure it's really worth trying to further strengthen the typing here. `custom_id_field` already validates that the value matches an existing key. I even added a test for that.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/329271cb1cd043023eac44518c43d1faf7b364c0/types/itemsjs/itemsjs-tests.ts#L114-L117